pkg/kotsadm/main.go

```diff
@@ -363,57 +423,85 @@ func ensureKotsadm(deployOptions types.DeployOptions, clientset *kubernetes.Clie
 	if deployOptions.License != nil {
 		// if there's a license, we write it as a secret and kotsadm will
 		// find it on startup and handle installation
-		if err := ensureLicenseSecret(&deployOptions, clientset); err != nil {
+		updated, err := ensureLicenseSecret(&deployOptions, clientset)
+		if err != nil {
 			return errors.Wrap(err, "failed to ensure license secret")
 		}
+
+		if updated {
+			restartKotsadmAPI = true
```

Comment:
I think secrets roll the pod when updated, but configmaps don't. We should rely on built-in k8s restarting and not restart on our own if all of the changes applied will result in k8s restarting the pod, or else we will likely end up with a weird scenario where it's restarted twice and fail

Reply:
both of these are optional and neither is mounted in the pod.  we try to read these in the automation package on startup.